### DISCUSSION TITLE
Dashboard ++ appeals but I have reservations.

### DISCUSSION THREAD

**Discussion by EricEWeir** [TfTHacker/DashboardPlusPlus#15] [Q&A]
I started exploring Obsidian with a top-level index similar to what you suggest. Two reservations:

(1) I do a lot of work on my phone. Not sure Dashboard ++ would work well there. Do you agree?

(2) I have avoided putting anything in my notes other than the actual note, e.g., anything YAML or Dataview related. Especially don’t want display of notes cluttered with that kind of thing. Will it be possible to clean files of this kind of thing once it’s been inserted?

And a third thing: Recently you wrote a review of the bookmark feature as a flexible and potentially powerful way of organizing notes. I take it that it would at least be redundant if not conflicting to try to use both bookmarks and Dashboard++ to analyze and organize notes.

**Comment by TfTHacker** [comment]
So nice to hear from you and your excellent and well-thought-out observations. Thank you. The following is just my opinion 🤓. I see we spoke on Medium, but thought it would be nice to add to the discussion here.

1) Dashboard++ works just fine on mobile and tablet. In fact, everything I do has an underlying requirement of good mobile support and I test that scenario thoroughly. BTW, the first image in the article shows desktop and mobile rendering of Dashboard++

2) Keep your notes as simple as possible is what I believe. So just a little YAML is needed for formatting. The rest is standard markdown lists. In the future, you can remove the YAML and not lose anything. I do use dataview for creating some simple file lists, but they are not crucial.

3) I think your bookmark question is significant. For sure there is some overlap here. What I can say is that I view my dashboards as more long-term, well optimized Maps of Content. One Dashboard leads to another, and to another. While the bookmark tree structure does accomplish the same thing, it doesn't allow for the visual benefit of the multiple columns of Dashboard++.

However, I do like bookmarks for working on "current things" - projects, searches, predefined graph views and so on. I am using bookmarks very much for "project" mode or specific repetitive tasks. While I can use Dashboard++ for this, bookmarks since its in a small pane makes some tasks much easier to do and not lose context.

Simply put: Dashboard++ has become my index into my long-term content, Bookmarks is used for focus on specific projects or for shortcuts to repetitive actions (search, opening multiple files, etc)

**Comment by EricEWeir** [comment]
Thanks for your thoughtful and helpful reply. I like the suggestion of distinguishing between “longer term” and “project” note taking and linking.As I indicated in my initial response, I began my experiment with Obsidian with a top-level index like yours. And there are a few lower level ones as well. So once I get the Dashboard ++ css snippet installed in my vault putting Dashboard++ to work will be, at least initially, pretty straightforward. A challenge for me is getting higher level notes linked to lower level ones when I am usually in the lower level ones. Perhaps the bookmarks feature/plug-in will be useful for that.———————————Eric Weir Atlanta, GA  USA“It has all been combustion.”- W.G. SebaldOn May 7, 2023, at 12:33 AM, TfTHacker ***@***.***> wrote:﻿So nice to hear from you and your excellent and well-thought-out observations. Thank you. The following is just my opinion 🤓. I see we spoke on Medium, but thought it would be nice to add to the discussion here.


Dashboard++ works just fine on mobile and tablet. In fact, everything I do has an underlying requirement of good mobile support and I test that scenario thoroughly. BTW, the first image in the article shows desktop and mobile rendering of Dashboard++


Keep your notes as simple as possible is what I believe. So just a little YAML is needed for formatting. The rest is standard markdown lists. In the future, you can remove the YAML and not lose anything. I do use dataview for creating some simple file lists, but they are not crucial.


I think your bookmark question is significant. For sure there is some overlap here. What I can say is that I view my dashboards as more long-term, well optimized Maps of Content. One Dashboard leads to another, and to another. While the bookmark tree structure does accomplish the same thing, it doesn't allow for the visual benefit of the multiple columns of Dashboard++.


However, I do like bookmarks for working on "current things" - projects, searches, predefined graph views and so on. I am using bookmarks very much for "project" mode or specific repetitive tasks. While I can use Dashboard++ for this, bookmarks since its in a small pane makes some tasks much easier to do and not lose context.
Simply put: Dashboard++ has become my index into my long-term content, Bookmarks is used for focus on specific projects or for shortcuts to repetitive actions (search, opening multiple files, etc)

**Comment by EricEWeir** [comment]
Well, I composed a note, somehow enlarged the screen, which made it unreadable, refreshed the page, and stupid me, my response was gone. Perhaps I can be more straightforward a second time around.

So, I predicted in my last response, that with my set of linked index files transitioning to the Dashboard++ setup would be relatives straightforward. Alas, it has not. I installed the css snippet and put the front matter at the fop of my top level index note. On reloading the top level index this is what it looked like:
 
![Screen Shot 2023-05-10 at 2 36 02 PM](https://github.com/TfTHacker/DashboardPlusPlus/assets/62710407/acd47757-c3b8-4386-8d6d-ceb6f047f25e)

Somehow the note didn't get interpreted by the Dashboard++ snippet. I played around trying to see if I discover what I'd missed, e.g., converting my top level list of linked index files to a list of unlinked files. That had no effect either:

![Screen Shot 2023-05-10 at 3 19 44 PM](https://github.com/TfTHacker/DashboardPlusPlus/assets/62710407/66795c52-49ce-4072-a8fb-d6302e042f58)

Just to verify that the snippet has been installed:
![Screen Shot 2023-05-10 at 3 22 03 PM](https://github.com/TfTHacker/DashboardPlusPlus/assets/62710407/9147f505-79fd-4ff4-94d7-a6e9ac20b0fc)

I don't think I'm gonna figure it out. I need help.

**Comment by TfTHacker** [comment]
You are close! So the first thing is that Dashboard++ formats a nested list. Right now you have a header followed by a list. What you want is a nested list like this:

- 0.0 Index
  - [1.0 Writing]()
  - more links
  - more links
- Next column heading
  - Link 1
  - link 2
  - Link 3

Finally, confirm in appearance that your dashboard.css snippet is enabled.

Let me know if you succeed with these additional steps.

**Reply by EricEWeir** [reply]
As I said in my last response, I was uncertain whether the items in my lists should be, or can be, links. Being uncertain I went ahead and created a list of items that are not links. In the process I made certain that the items under each of the headings corresponded to the items in my lists of links. So if I should've created a list of links I can convert them without much difficulty. Here's what it looks like:

![Screen Shot 2023-05-11 at 3 06 40 PM](https://github.com/TfTHacker/DashboardPlusPlus/assets/62710407/b4aebe13-854b-4294-8d47-8f39a29bbece)

I haven't put the front matter in yet. I'm unclear whether each of the headings with subtopics will become separate file, if so whether they too will need the front matter, and whether it should be identical to the main document. I assume the list of headers and subtopics will be a file, but unclear it should be named, whether the name will be, should be an element in the list.

**Comment by EricEWeir** [comment]
Thanks. (I feel like I should know your name.)I’m writing before an early morning run in a lovely place far from. So will report results later.The links in that index are my “next headers.” Each of them with a list of linked notes.One thing I’m not clear about: should the items in my lists be, or can they be, links? Or does the Dashboard++ snippet take care of linking from items in the lists to the notes they refer to?More later,———————————Eric Weir Atlanta, GA  USA“It has all been combustion.”- W.G. SebaldOn May 11, 2023, at 5:58 AM, TfTHacker ***@***.***> wrote:﻿You are close! So the first thing is that Dashboard++ formats a nested list. Right now you have a header followed by a list. What you want is a nested list like this:

0.0 Index

1.0 Writing
more links
more links


Next column heading

Link 1
link 2
Link 3



Finally, confirm in appearance that your dashboard.css snippet is enabled.
Let me know if you succeed with these additional steps.

**Comment by Kmfernan5** [comment]
Based on the discussion, are these the specific changes TfTHacker recommended to get Dashboard++ working properly with Eric's notes:

- Structure the top-level index as a nested list with column headers, like:

0.0 Index 
1.0 Writing
- Item1 
- Item2

2.0 Next Header 
- Item1
- Item2

- The items under each header should correspond to the existing linked index notes Eric already has.

- Make sure the dashboard.css snippet is enabled in Obsidian to get the Dashboard++ styling.

- The list items can be links to notes, or just text - Dashboard++ will handle linking them. 

- Remove the header Eric had above his top level list, since Dashboard++ needs the nested list structure.

- Each column header and list will become a separate file when rendered by Dashboard++.

- Add the Dashboard++ frontmatter to the main index file.


I'm other words...

TfTHacker has suggested several changes to optimize Dashboard++'s performance with Eric's notes:

1. **Nested List Structure**: Reframe the top-level index as a nested list with column headers. Each header should have relevant items underneath, which can correspond to Eric's existing linked index notes.

2. **Enable CSS Snippet**: Make sure the dashboard.css snippet is active in Obsidian for the Dashboard++ aesthetic.

3. **Links and Text**: The items in the list can either be links to notes or simply plain text. Dashboard++ will manage the linking.

4. **Remove Extra Header**: Delete the header above Eric's top-level list as Dashboard++ requires the nested list structure.

5. **Rendered Files**: Be aware that each column header and list will turn into a separate file when Dashboard++ renders it.

6. **Dashboard++ Frontmatter**: Don't forget to add the Dashboard++ frontmatter to the main index file.

In essence, the key solutions involve the construction of a nested list, CSS activation, and the removal of the superfluous header.
